feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -1,155 +1,163 @@
+import { Input, Output, EventEmitter, Directive } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorAttributeQuantityService } from '../../quantity';
+import { ConfiguratorUiKeyGeneratorService } from './configurator-ui-key-generator.service';
 
-/**
- * Service to provide unique keys for elements on the UI and for sending to configurator
- */
-
+@Directive()
+// eslint-disable-next-line @angular-eslint/directive-class-suffix
 export class ConfiguratorAttributeBaseComponent {
```

Comment:
This is `abstract` class?

Reply:
No, it is a base component with the common logic for both single and multi-selection attribute types

Reply:
yes, but there is no instance of it, right? and there is `@Directive()`.

Reply:
sorry, yes, there is not instance of the base component. Both single and multi-selection components extend the base component

Reply:
I think, it should be `abstract` class.

Reply:
Done!

Reply:
Great point! :)